### PR TITLE
[AIRFLOW-1618] Add feature to create GCS bucket

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1618

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
- Added `create_bucket` method to `gcs_hook` and created corresponding operator `GoogleCloudStorageCreateBucket`
- Added tests
- Added documentation

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
`GoogleCloudStorageCreateBucketTest`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/operators/gcs_operator.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class GoogleCloudStorageCreateBucket(BaseOperator):
```
Comment (fenglu-db) [inline airflow/contrib/operators/gcs_operator.py]: s/GoogleCloudStorageCreateBucket/GoogleCloudStorageCreateBucketOperator

File: airflow/contrib/operators/gcs_operator.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                 google_cloud_storage_conn_id='google_cloud_storage_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+        super(GoogleCloudStorageCreateBucket, self).__init__(*args, **kwargs)
+        self.bucket_name = bucket_name
+        self.storage_class = storage_class
+        self.location = location
+        self.project_id = project_id
+        self.labels = labels
```
Comment (fenglu-db) [inline airflow/contrib/operators/gcs_operator.py:97]: mind adding a default airflow system labe? e.g., https://github.com/apache/incubator-airflow/blob/master/airflow/contrib/operators/dataflow_operator.py#L116

**Review by fenglu-db — COMMENTED** [review]
Thanks @kaxil, a few minor comments.

File: airflow/contrib/operators/gcs_operator.py (hunk 2 above)
Comment (kaxil) [reply airflow/contrib/operators/gcs_operator.py:97]: @fenglu-g Can do that. But do you want me to replace `labels=None` to `labels={'airflow-version': 'v' + version.replace('.', '-').replace('+', '-')})` or update whatever labels are passed with `labels.update({'airflow-version': 'v' + version.replace('.', '-').replace('+', '-')}))`

My only concern is some companies have a strict policy for using labels to track their pricing and already have a naming convention set for their labels. 

What's your suggestion? Should we be passing the `airflow-version` label if no labels are passed or just add this key to whatever labels they pass?

Comment (kaxil) [reply airflow/contrib/operators/gcs_operator.py:97]: Or do you want me to add something like:
```python
if self.labels is not None:
    self.labels.update(
        {'airflow-version': 'v' + version.replace('.', '-').replace('+', '-')}
    )
```

**Comment by Fokko** [comment]
Could you rebase on master? CI has been fixed.

**Comment by Fokko** [comment]
Let me know when LGTM @kaxil @fenglu-g

File: airflow/contrib/operators/gcs_operator.py (hunk 2 above)
Comment (fenglu-db) [reply airflow/contrib/operators/gcs_operator.py:97]: The latter, a combination of both airflow system labels and customer supplied labels. The idea is to measure the impact of Airflow.

Comment (kaxil) [reply airflow/contrib/operators/gcs_operator.py:97]: Done

**Comment by kaxil** [comment]
@Fokko LGTM now.

**Comment by kaxil** [comment]
I still need to change the test which is causing the CI to fail. Will do that late today.

**Comment by kaxil** [comment]
@Fokko I have fixed the tests and LGTM. This can now be merged.

**Pull request closed without merging**